src/interventions.c

```diff
@@ -528,6 +528,174 @@ void intervention_test_order( model *model, individual *indiv, int time )
+	} else
+		set_vaccine_status( indiv, VACCINE_NO_PROTECTION, NO_EVENT );
+
+	return TRUE;
+}
+
+/*****************************************************************************************
+*  Name:		intervention_vaccinate_by_idx
+*  Description: Vaccinate an individual
+*
+*  Arguments:   model           - pointer to the model
+*  				idx             - idx of person to vaccinate
+*  				vaccine_type    - whether the vaccine is FULL or SYMTOMS_ONLY
+*  				efficacy        - probability the person is successfully vaccinate
+*  				time_to_protect - delay before it takes effect
+*  				vaccine_protection_period - length of time the vaccine provides protection
+*
+*  Returns:		1 if vaccinated 0 if not
+******************************************************************************************/
+short intervention_vaccinate_by_idx(
+	model *model,
+	long idx,
+	short vaccine_type,
+	double efficacy,
+	short time_to_protect,
+	short vaccine_protection_period
+)
+{
+	return intervention_vaccinate( model, &(model->population[idx]), vaccine_type, efficacy, time_to_protect, vaccine_protection_period );
+}
+
+
+/*****************************************************************************************
+*  Name:		intervention_vaccinate_age_groups
+*  Description: Vaccinate a fraction of each age group
+*
+*  Arguments:   model           - pointer to the model
+*  				fractions       - the fraction of the population in each age group to vaccinate
+*  				vaccine_type    - whether the vaccine is FULL or SYMTOMS_ONLY
+*  				efficacy        - probability the person is successfully vaccinate
+*  				time_to_protect - delay before it takes effect
+*
+*  Returns:		long - the total number of people vaccinated
+******************************************************************************************/
+long intervention_vaccinate_age_group(
+	model *model,
+	double fractions[ N_AGE_GROUPS ],
+	short vaccine_type,
+	double efficacy,
+	short time_to_protect,
+	short vaccine_protection_period,
+	long n_vaccinated[ N_AGE_GROUPS ]
+)
+{
+	long n_to_vaccinate[ N_AGE_GROUPS ];
+	long total_to_vaccinate, total_vaccinated, pdx;
+	short age;
+	individual *indiv;
+
+	total_to_vaccinate = 0;
+	total_vaccinated  = 0;
+	for( age = 0; age < N_AGE_GROUPS; age++ )
+	{
+		n_to_vaccinate[ age ] = round( model->n_popualtion_by_age[ age ] * fractions[ age ] );
+		n_vaccinated[ age ]   = 0;
+		total_to_vaccinate   += n_to_vaccinate[ age ];
+	}
+
+	for( pdx = 0; pdx < model->params->n_total; pdx++ )
+	{
+		indiv = &(model->population[pdx]);
+		age   = indiv->age_group;
+
+		if( n_to_vaccinate[ age ] == n_vaccinated[ age ] )
+			continue;
+		if( indiv->vaccine_status != NO_VACCINE )
+			continue;
+
+		intervention_vaccinate( model, indiv, vaccine_type, efficacy, time_to_protect, vaccine_protection_period );
+		n_vaccinated[ age ]++;
+		total_vaccinated++;
+
+		if( total_to_vaccinate == total_vaccinated )
+			break;
+	}
+
+	return total_vaccinated;
+}
+
```

Comment:
Checked this earlier and is still possible to vaccinate dead individuals though they remain dead. Wondering if there are other groups we may wish to avoid vaccinating too such as those currently in critical care,  perhaps by providing some `EVENT_TYPES` enums to ignore by default which could also be modified if needed?

Reply:
I've added a check that a person is not dead or in hospital before vaccinating and return false from `intervention_vaccinate()` if this is the case.
When running a vaccine schedule, I check the return value of `intervention_vaccinate()` and if returns False then don't count that attempt of a vaccination.